### DISCUSSION TITLE
pyvisfile installation failure with SILO

### DISCUSSION THREAD

**Discussion by Chunhui92** [inducer/pyvisfile#30] [Q&A]
Dear developers,

I am a Phd student in CEA France, recently I tried to install pyvisfile to read .silo files. 

The silo version is silo-4.10-bsd, the pyvisfile version is pyvisfile-2022.1. I followed exactly the steps in [pyvisfile documentation](https://documen.tician.de/pyvisfile/installing.html#with-silo-capability), while the step failed with the error information: 
_"
lib/python3.8/site-packages/numpy/core/include/numpy/npy_1_7_deprecated_api.h:17:2: warning: #warning "Using deprecated NumPy API, disable it with " "#define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION" [-Wcpp]
 #warning "Using deprecated NumPy API, disable it with " \
  ^
src/wrapper/wrap_silo.cpp:15:18: fatal error: silo.h: No such file or directory
compilation terminated.
error: command 'gcc' failed with exit status 1
"_

I've re-checked the _.aksetup-defaults.py_ and _siteconf.py_ files, the pathes to SILO/include and SILO/lib are ok, and the _silo.h_ is located in SILO/include.

Can you give me some clue about this error?

Sincerely,

**Comment by inducer** [comment]
Could you copy the output of the entire build process and put it in a gist? Also paste your `siteconf.py`. (You only need one of `.aksetup-defaults.py` and `siteconf.py`. Delete the other.)

**Reply by Chunhui92** [reply]
Some updates: 
1. The build process was success, while error occured when importing pyvisfile.silo (see step 4 and 5).
2. According to the error message, I manually configured pyvisfile with --use-silo, then error occured again (see step 6).

The outputs are put in the gist: https://gist.github.com/Chunhui92/917a4110eb8328eba96971d8b692a8aa
The siteconf.py is:
BOOST_INC_DIR = []
BOOST_LIB_DIR = []
BOOST_COMPILER = 'gcc43'
USE_SILO = True
SILO_INC_DIR = ['/home/pool/include']
SILO_LIB_DIR = ['/home/pool/lib']
SILO_LIBNAME = ['siloh5']
CXXFLAGS = []

**Reply by inducer** [reply]
For
```
ModuleNotFoundError: No module named 'pyvisfile.silo._internal'
```
try
```
ldd /home/dangchunhui/silo-4.10-bsd/pyvisfile-2022.1/pyvisfile/silo/_internal.so
```
If that shows that `libsilo` is not found, you need need to set `LD_LIBRARY_PATH`.

**Reply by Chunhui92** [reply]
It shows _ldd: /home/dangchunhui/silo-4.10-bsd/pyvisfile-2022.1/pyvisfile/silo/_internal.so: No such file or directory
_
Apparently the _internal.so file was not created during the build process, while I followed exactly the steps listed in pyvisfile documentation. Do you have any clue about this? Thanks!

**Reply by Chunhui92** [reply]
Sorry but are you still working on this problem? Appreciate for your feedbacks.
